File: backend/core-services/eligibility-service/src/main/java/com/govtech/eligibility/domain/model/Eligibility.java
package com.govtech.eligibility.domain.model;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

import lombok.Builder;

@Builder
public record Eligibility(UUID id,
        UUID calculationId,
        Instant calculatedAt,
        String subject,
        String aidCode,
        String aidName,
        EligibilityStatus status,
        String reason,
        BigDecimal estimatedAmount,
        String estimatedAmountLabel) {
}

File: frontend/portal-web/src/features/eligibility/types/eligibility.ts
export interface Eligibility {
  aidCode: string;
  aidName: string;
  description: string;
  category: string;
  status: "ELIGIBLE" | "NOT_ELIGIBLE" | "PENDING";
  estimatedAmount: number | null;
  estimatedAmountLabel: string;
  reason: string;
  canApply: boolean;
  actionLabel: string;
  icon: string;
  applicationId?: string | null;

}